core/src/Kokkos_Half.hpp

```diff
@@ -226,9 +226,19 @@ class alignas(FloatType) floating_point_wrapper {
 #if defined(_WIN32) && defined(KOKKOS_ENABLE_CUDA)
   KOKKOS_FUNCTION
   floating_point_wrapper(const floating_point_wrapper& rhs) : val(rhs.val) {}
+
+  KOKKOS_FUNCTION
+  floating_point_wrapper& operator=(const floating_point_wrapper& rhs) {
+    val = rhs.val;
+    return *this;
+  }
```

Comment:
Please give some context for this addition.

Reply:
The warning requires us to define a copy assignment operator if a copy constructor is defined and since we can't make the latter one `default`, it seems reasonable also not to make the former one `default`.

Reply:
It looks like this was added to surpress a compiler warning about copy assignment operators. We could consider using pragma to supress that warning instead. But, as long as the half precision unit tests pass and are extended (or verified) to cover this operator, I think the change is fine. It would also be good to add a comment above these added lines explaining why they are needed.

Reply:
> It looks like this was added to surpress a compiler warning about copy assignment operators. 

This pull request is about compatibility with a new warning flag that flags exactly set. 🙂
I don't see a good reason why we would try to suppress the warning we are adding here. Note that we could simplify all this significantly if we dropped the `volatile` overloads.

> We could consider using pragma to supress that warning instead. But, as long as the half precision unit tests pass and are extended (or verified) to cover this operator, I think the change is fine.

We get a bunch of compile errors if we tried to delete the copy assignment operator.

Reply:
I don't think we have any reason to keep the `volatile` overloads around any more